Update SSO prereqs for consistency and include Teleport Cloud

stevenGravy opened the pull request:
- Made prerequisites consistent
- tctl enterprise download like database access guides

stevenGravy committed:
Consistent prerequisites and Teleport cloud

stevenGravy committed:
Modified to make cloud example

ptgott commented on docs/pages/enterprise/sso.mdx:
````diff
@@ -67,19 +67,43 @@ The following connectors are supported:
+  version: v2
+  ```
+
+  Create a resource:
````
```suggestion
  Create the resource:
```
Since we've established which resource we are creating

ptgott commented on docs/pages/enterprise/sso/adfs.mdx:
```diff
@@ -22,6 +22,15 @@ like:
 
+## Prerequisites
+
+- Either an Enterprise version of Teleport (downloaded via the [Enterprise dashboard](https://dashboard.gravitational.com/web/login)) or a Teleport Cloud account (sign up for a [free trial](https://goteleport.com/signup)).
```
- Does this mean that the user needs a running Auth/Proxy service? 

- I think we can adjust the visibility of this sentence based on scope, i.e., so Cloud users don't see Enterprise-specific information. (Similar to this section: https://goteleport.com/docs/access-controls/getting-started/#prerequisites)

ptgott commented on docs/pages/enterprise/sso/gitlab.mdx:
```diff
@@ -15,6 +15,15 @@ like:
 
+## Prerequisites
+
+- Either an Enterprise version of Teleport (downloaded via the [Enterprise dashboard](https://dashboard.gravitational.com/web/login)) or a Teleport Cloud account (sign up for a [free trial](https://goteleport.com/signup/)).
```
Same comments as previously re: this Prerequisites item

ptgott commented on docs/pages/enterprise/sso/oidc.mdx:
```diff
@@ -20,6 +20,16 @@ administrators to define policies like:
 
+## Prerequisites
+
+- Either an Enterprise version of Teleport (downloaded via the [Enterprise dashboard](https://dashboard.gravitational.com/web/login)) or a Teleport Cloud account (sign up for a [free trial](https://goteleport.com/signup)).
```
Same comments as previously re: this Prerequisites item

ptgott commented on docs/pages/enterprise/sso/okta.mdx:
```diff
@@ -22,6 +22,15 @@ like:
 
+## Prerequisites
+
+- Either an Enterprise version of Teleport (downloaded via the [Enterprise dashboard](https://dashboard.gravitational.com/web/login)) or a Teleport Cloud account (sign up for a [free trial](https://goteleport.com/signup)).
```
Same comments as previously re: this Prerequisites item

ptgott commented on docs/pages/enterprise/sso/one-login.mdx:
```diff
@@ -22,6 +22,15 @@ like:
 
+## Prerequisites
+
+- Either an Enterprise version of Teleport (downloaded via the [Enterprise dashboard](https://dashboard.gravitational.com/web/login)) or a Teleport Cloud account (sign up for a [free trial](https://goteleport.com/signup)).
```
Same comments as previously re: this Prerequisites item

ptgott commented on docs/pages/includes/enterprise/ent-user-prereqs.mdx:
```diff
@@ -0,0 +1,31 @@
+<TabItem
+  scope={["enterprise"]} label="Self-Hosted">
+
+- Access to the machine hosting the Auth Service
```
Should we mention that you can also manage a self-hosted cluster remotely with `tctl`? This might be easier for the user if accessing the machine hosting the Auth Service is difficult. Or does this require elevated permissions that you can only obtain by running `tctl` on the same machine as the Auth Service?

ptgott commented on docs/pages/includes/enterprise/ent-user-prereqs.mdx:
```diff
@@ -0,0 +1,31 @@
+
+- Access to the machine hosting the Auth Service
+
+- Maintaining SAML/OIDC auth connectors via the desktop CLI requires enterprise `tctl` admin tool version >= (=teleport.version=),
```
```suggestion
- Maintaining SAML/OIDC auth connectors via the desktop CLI requires the enterprise `tctl` admin tool version >= (=teleport.version=),
```

ptgott replied:
It's not clear to me in this tab what the relationship is between this bullet and the previous one. Should the user have both access to the Auth Service host _and_ the Enterprise version of `tctl` installed locally?

ptgott commented on docs/pages/includes/enterprise/ent-user-prereqs.mdx:
```diff
@@ -0,0 +1,31 @@
+<TabItem scope={["cloud"]}
+  label="Teleport Cloud">
+
+- The `tctl` client Admin tool version >= (=cloud.version=).
```
```suggestion
- The `tctl` admin client version >= (=cloud.version=).
```

ptgott commented on docs/pages/includes/enterprise/ent-user-prereqs.mdx:
```diff
@@ -0,0 +1,31 @@
+
+- The `tctl` client Admin tool version >= (=cloud.version=).
+
+  You can download these from [Teleport Cloud Downloads](/docs/cloud/downloads).
```
```suggestion
  You can download this from [Teleport Cloud Downloads](/docs/cloud/downloads).
```

stevenGravy committed:
spelling mistake

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

stevenGravy committed:
missing period

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

stevenGravy committed:
verbiage

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

stevenGravy committed:
verbiage

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

stevenGravy committed:
verbiage

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

stevenGravy committed:
cloud and self-host default auth example

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

stevenGravy committed:
Modified pre-req to show required enterprise or cloud installed with scope

stevenGravy committed:
verbiage

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

stevenGravy committed:
Updated github to match other ssos.  tctl desktop connection

stevenGravy committed:
Makes consistent across github and ent ssos

stevenGravy committed:
lint fixes

ptgott commented on docs/pages/includes/enterprise/ent-user-prereqs.mdx:
```diff
@@ -0,0 +1,30 @@
+<TabItem
+  scope={["enterprise"]} label="Self-Hosted">
+- Installed Enterprise version of Teleport (downloaded via the [Enterprise dashboard](https://dashboard.gravitational.com/web/login)).
+- Maintain via the `tctl` enterprise admin tool via the Desktop (recommended) or have administrative access on the Teleport auth service machine.
```
I think we can condense this and the following bullets into a single bullet, since they both instruct you to download `tctl`. Also, I think we can make it a lot simpler by borrowing from the current version of `edition-prereqs-tabs.mdx` (this example also includes `tsh`, which I think we should mention if we're going to show `tsh` commands):

```md
- The `tctl` admin tool and `tsh` client tool version >= 
(=teleport.version=), installed on your local machine.

You can download these by visiting the [customer portal](https://dashboard.gravitational.com/web/login).
```

We can then change `tctlconnection.mdx` to make it explicit that you should run the `tctl` command on your local machine.

ptgott commented on docs/pages/includes/enterprise/ent-user-prereqs.mdx:
````diff
@@ -0,0 +1,30 @@
+  $ tctl version
+  # Teleport v(=cloud.version=) go(=teleport.golang=)
+  ```
+- Teleport role with access to maintaining `saml` and `oidc` resources.  This is available in the default `editor` role.
````
```suggestion
- Teleport role with access to maintaining `saml` and `oidc` resources. This is available in the default `editor` role.
```
Also, the bullet isn't rendering for some reason.

ptgott commented on docs/pages/includes/enterprise/ent-user-prereqs.mdx:
```diff
@@ -0,0 +1,30 @@
+<Tabs>
```
If this partial is only going to be used in SSO guides, should we move it to the `docs/pages/includes/sso` directory?

ptgott commented on docs/pages/includes/sso/tctlconnection.mdx:
```diff
@@ -0,0 +1,40 @@
+<Details 
```
How would you describe the division of labor between this partial and `tctl.mdx`?

stevenGravy committed:
verbiage update

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

stevenGravy committed:
verbiage update for adfs prereq

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

stevenGravy committed:
okta prereq verbiage updated

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

stevenGravy committed:
remove spaces

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

stevenGravy committed:
verbiage change

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

stevenGravy committed:
Corrected OIDc prereq and fixed bullet in include

stevenGravy commented on docs/pages/enterprise/sso/oidc.mdx:
```diff
@@ -20,6 +20,15 @@ administrators to define policies like:
+## Prerequisites
+
+- Admin access to the SSO/IdP being integrated.
+- Users with groups/roles assigned.
```
Yes, updated language

stevenGravy committed:
Removed specific github content from editition-prereqs-tabs and moved to github-sso

stevenGravy commented on docs/pages/includes/edition-prereqs-tabs.mdx:
```diff
@@ -21,42 +21,36 @@ files in partials, this partial uses relative URL paths instead.
 
+- Maintain via the `tctl` enterprise admin tool via the Desktop (recommended) or have administrative access on the Teleport auth service machine.
+
+- Teleport role with access to maintaining `github` resources for using `tctl` from the Desktop. This is available in the default `editor` role.
```
removed and placed in github-sso only

stevenGravy committed:
verbiage change

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

stevenGravy committed:
Addressed review items

stevenGravy committed:
Combined tctl instructions

stevenGravy commented on docs/pages/includes/edition-prereqs-tabs.mdx:
````diff
@@ -21,42 +21,36 @@ files in partials, this partial uses relative URL paths instead.
-  # Teleport v(=teleport.version=) go(=teleport.golang=)
   ```
-
+  Teleport role with access to maintaining `saml` and `oidc` resources.  This is available in the default `editor` role.
````
addressed

stevenGravy commented on docs/pages/includes/edition-prereqs-tabs.mdx:
````diff
@@ -21,42 +21,36 @@ files in partials, this partial uses relative URL paths instead.
-  $ tsh version
-  # Teleport v(=cloud.version=) go(=teleport.golang=)
   ```
+- Teleport role with access to maintaining `saml` and `oidc` resources.  This is available in the default `editor` role.
````
addressed

stevenGravy commented:
@ptgott Please review updates.  This should be more consistent and direct on the approach in the enterprise SSO and GitHub.

ptgott commented:
Hi @stevenGravy, sorry for the delay in reviewing this. Would you mind resolving the conflicts in `docs/pages/enterprise/sso/gitlab.mdx`? Once that's done, I'll make time to review tomorrow. Thanks!

ptgott committed:
Edit SSO guides (#12964)

Edit the SSO introduction:

- Add examples of all connectors so users don't need to find them
  in the examples directory
- Add subheadings to the "Configuring SSO" section for clarity
- Minor grammar/style edits

Specific SSO guides:

- Use the commercial-prereqs-tabs partial in Prerequisites sections,
  since this partial already exists, rather than ent-user-prereqs.mdx.
  This partial is more specific about requiring a running Teleport
  cluster, and links to the Getting Started guide for the Enterprise
  edition.

- Minor tweaks to the scoped information shown in the
  samlauthentication.mdx and oidcauthentication.mdx partials

- Restored the original wording of edition-prereqs-tabs.mdx, which
  is more specific about how to get started with a running Teleport
  cluster on different editions, and includes instructions for tsh.

stevenGravy committed:
remove extra line

stevenGravy committed:
Remove extra line

ptgott committed:
Move the Okta connector into sso.mdx

This was triggering a full build in the CI pipeline, which was
causing issues merging this branch
